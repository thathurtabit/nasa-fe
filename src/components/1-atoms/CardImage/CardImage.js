import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { CSSTransition } from 'react-transition-group';
import IMG from './CardImage.styled';
import LoadingSmall from '../LoadingSmall/LoadingSmall';
import { loadDelay } from '../../../utils/constants/constants';

export default class CardImage extends Component {
  constructor(props) {
    super(props);
    this.state = { isLoading: true };
  }

  componentDidMount() {
    // Only show images once load has complete
    // Note: Must be a prettier way of doing this
    const img = new Image();
    const { url } = this.props;
    img.src = url;
    img.onload = () => this.setState({ isLoading: false });
  }

  render() {
    const { isLoading } = this.state;
    const { title, url, thumb } = this.props;
    return (
      <Fragment>
        {isLoading && <LoadingSmall />}
        {!isLoading && (
          <CSSTransition
            in={!isLoading}
            classNames="thumb"
            appear
            timeout={loadDelay}
          >
            <IMG className="thumb" src={url} alt={title} thumb={thumb} />
          </CSSTransition>
        )}
      </Fragment>
    );
  }
}

CardImage.propTypes = {
  thumb: PropTypes.bool,
  url: PropTypes.string.isRequired,
  title: PropTypes.string,
};

CardImage.defaultProps = {
  thumb: false,
  title: 'Card',
};
