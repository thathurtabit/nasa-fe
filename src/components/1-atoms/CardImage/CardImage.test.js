import React from 'react';
import { shallowToJson } from 'enzyme-to-json';
import 'jest-styled-components';
import { delay } from 'bluebird';
import { shallowWithTheme } from '../../../utils/test_config/testHelpers';

import CardImage from './CardImage';

const title = 'test title';
const url =
  'https://d1xkhapf8f3lxw.cloudfront.net/api/images/CardShop/57/product/tw902';

describe('CardImage', () => {
  it('should render correctly', () => {
    const element = shallowWithTheme(<CardImage url={url} title={title} />);
    expect(shallowToJson(element)).toMatchSnapshot();
  });
});
