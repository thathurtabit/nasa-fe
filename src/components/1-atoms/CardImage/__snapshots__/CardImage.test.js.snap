// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardImage should render correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
